tests/integration/commands/dev/dev.config.test.js

```diff
@@ -451,6 +451,29 @@ describe.concurrent('commands/dev/config', () => {
     })
   })
 
+  test('will prefer a redirect to a function that matches the path', async (t) => {
```

Comment:
I'm not a fan of this, and I don't know how to reproduce it better.

When I checked locally, the behavior was fixed by the changes in `src/utils/proxy.ts`. However, if I undo the changes in proxy, this test doesn't fail, and I don't understand why.